### ISSUE TITLE
[$1000] Pressing space bar while details page is opened, scrolls up the reports

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15631]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Click on an avatar in a money request 
2. After the details page has opened, press space bar

## Expected Result
Nothing should happen

## Actual Result
Chats are scrolled up

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.78-0
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/222602809-9e824b5e-3df6-4689-b11f-6c869a0ec559.mov

https://user-images.githubusercontent.com/43996225/222603060-e7762a4b-cc57-4aca-9567-6bb1ac209f2f.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @esh-g 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1677578613083529

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017ddd7486addffc52</li>
        <li>Upwork Job ID: 1632967018098249728</li>
        <li>Last Price Increase: 2023-08-08</li>
<li>Automatic offers: </li>
<ul>
<li>fedirjh | Reviewer | 26060321</li>
<li>esh-g | Reporter | 26060323</li>
</ul></li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @stephanieelliott (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
@stephanieelliott Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by stephanieelliott** [comment]
Yeah this is weird behavior. Made a small tweak to the issue description (changed `report` to `request money`), looks all set to go to a contributor for a fix!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~017ddd7486addffc52

**Comment by MelvinBot** [comment]
Current assignee @stephanieelliott is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @rushatgabhane (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @joelbettner (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by tienifr** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After the details page has opened, press space bar will scroll the chats up.

### What is the root cause of that problem?
This happens because the current `StackView` implementation that we have does not have focus traps. When we press space or up/down arrows, the keyboard event will be triggered on the body element and cause the chat to scroll up. (Chat scrolling up is normal behavior, just that it should not happen when a modal is opened on top)

### What changes do you think we should make in order to solve the problem?

We need to add focus trap to the pages in the right sidebar, we can use this package https://github.com/focus-trap/focus-trap-react, and add the `<FocusTrap>` element in here https://github.com/Expensify/App/blob/f1769bc380c25d28e1c0bc75d449f2a1740f864b/src/pages/DetailsPage.js#L101

### What alternative solutions did you explore? (Optional)
We can add the focus trap in the `react-navigation` itself, or wrap all our `StackView` with the focus trap instead of adding it in individual pages.

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by you1996** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When the details page is open, pressing on space bar will cause scroll down/up (based on the initial position).

### What is the root cause of that problem?

As we can see, the chats section has a stack of components, with the details modal being the most recent one. Any scroll event triggered by the user (in this case, by pressing the space key) will be applied to the body element since the details page does not have a body element of its own. This means that the scroll event will automatically be applied to the parent component, which is a normal and expected behavior.

### What changes do you think we should make in order to solve the problem?

As the problem is caused by normal behavior, I suggest simply disabling scrolling using the space key when the details page is open. Adding more complexity, packages, or changing the stack view behavior is not the best solution. Therefore, we need to add an event listener on the details page that listens for a "keydown" event with a code of 32 (space) in the componentDidMount(), and prevent the default action on the body, we should also add a little function to handle the event.

```
componentDidMount() {
       ...
       document.addEventListener("keydown", this._handleKeyDownEvent);
}
```
### What alternative solutions did you explore? (Optional)

We also considered an alternative solution of disabling scrolling using the space key on all pages in the app by adding the listener on the app component. This ensures that the issue will not occur again.

**Comment by MelvinBot** [comment]
📣 @you1996! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by you1996** [comment]
Contributor details
Your Expensify account email: youssribentaghalline@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01b676ad45ba72ff4f

**Comment by pradhankukiran** [comment]
> Is it expected that I am able to see this happening without having to open the details page, @rushatgabhane?

Yes, when the text field is not focussed.

**Comment by pradhankukiran** [comment]
Contributor details
Your Expensify account email: pradhankukiran@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01b12157cda25a9ae3

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by pradhankukiran** [comment]
Not only pressing _Space Bar_ scrolls up the reports, pressing _Arrow Up_ scrolls down the page & pressing _Arrow Down_ scrolls up the page while details page is opened or when the text field is not focused.

**Comment by stephanieelliott** [comment]
Hey @rushatgabhane can you please review these proposals? https://github.com/Expensify/App/issues/15631#issuecomment-1457663588 and https://github.com/Expensify/App/issues/15631#issuecomment-1459070158

**Comment by s77rt** [comment]
It may be a good call to hold this for [navigation reboot](https://github.com/Expensify/App/issues/11768).

**Comment by MelvinBot** [comment]
@joelbettner, @rushatgabhane, @stephanieelliott Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by stephanieelliott** [comment]
Thanks for calling that our @s77rt! Chatted with @joelbettner and we are going to hold this one on the navigation reboot. Adding this to the tracking issue and putting on `[HOLD]`.

**Comment by s77rt** [comment]
@stephanieelliott Let's keep the issue open

**Comment by stephanieelliott** [comment]
Oops, yeah didn't man to close it 🙃, thanks @s77rt! I agree, let's keep it open!

**Comment by joelbettner** [comment]
Any word on the status with this bug? Still waiting on the navigation reboot?

**Comment by stephanieelliott** [comment]
Still waiting on the nav reboot! Latest status there: 

> Implementation still going strong. C+ has helped a lot with testing and identifying some bugs, we are now focusing on getting the first PR which addresses 3 issues merged in the navigation-refactor branch and move onto the other issues.

**Comment by joelbettner** [comment]
Popping in here for another status check on this.

**Comment by joelbettner** [comment]
Confirmed, I was also able to reproduce.

@stephanieelliott given that it has been a few months, how do we feel about opening this up to be worked on now, or should we still wait?

**Comment by stephanieelliott** [comment]
https://github.com/Expensify/App/pull/19263 was deployed to staging a few hours ago. Once it hits prod we will take this off hold.

**Comment by stephanieelliott** [comment]
The nav refractor has hit production, so reopening this one now. 

We had some proposals from when this was originally opened in March, @rushatgabhane can you please review? https://github.com/Expensify/App/issues/15631#issuecomment-1457663588 and https://github.com/Expensify/App/issues/15631#issuecomment-1459070158

**Comment by stephanieelliott** [comment]
> We had some proposals from when this was originally opened in March, @rushatgabhane can you please review? https://github.com/Expensify/App/issues/15631#issuecomment-1457663588 and https://github.com/Expensify/App/issues/15631#issuecomment-1459070158

Bump @rushatgabhane, can you please review the proposals?

**Comment by stephanieelliott** [comment]
Reapplying the `Bug` label to get another BZ member on this while I am OOO til July 30. Thanks @kevinksullivan,  I'll grab this back from you when I return!

**Comment by joelbettner** [comment]
@tienifr any word on this?

**Comment by joelbettner** [comment]
I'm going to be OOO for a good portion of next week and the week after that. Feel free to re-assign a different engineer if need be.

**Comment by kevinksullivan** [comment]
Bumping this in slack

https://expensify.slack.com/archives/C049HHMV9SM/p1687908425385029?thread_ts=1677578613.083529&cid=C049HHMV9SM

**Comment by esh-g** [comment]
@rushatgabhane I think this issue was present even before the navigation refactor (Mar 3)

**Comment by kevinksullivan** [comment]
Let's take this to slack ^

**Comment by joelbettner** [comment]
I'm going to unassign myself while this is being hashed out in Slack. Feel free to re-assign me afterwards if needed.

**Comment by kevinksullivan** [comment]
Discussing in slack

**Comment by rushatgabhane** [comment]
@kevinksullivan could you please attach the link, i can't seem to find it

**Comment by kosmydel** [comment]
Hey, I'm from SWM.

I investigated this issue for a moment.

---

The `FocusTrap` solution sounds great and it'll probably solve another bug that I found during the investigation:

**Bug:** It is possible to select the message input bar and send a message in the background, while the details page is opened. I attach a video below.

<details>
  <summary>Sending messages in the background</summary>
 
https://github.com/Expensify/App/assets/104823336/a991a7ea-171d-4c8e-91ee-c0798b1226bd
</details>

So we should confirm that the focus trap solves this issue as well.

---

If you need any assistance with the implementation or anything, I am ready to help.

**Comment by kevinksullivan** [comment]
@rushatgabhane can you take a look at @tienifr 's previous comment?

**Comment by kosmydel** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
When the RHP is open, the user can access and do things in the background.

From my observations, the user can in the background:
- press the space to scroll the report page (described in this issue),
- navigate using `tab`/`shift+tab`:
- send a message in the background,
- change reports,
- attach files to the report,
- probably a lot of different things...

### What is the root cause of that problem?
Users can shift their focus outside the modal, allowing them to interact with elements in the background instead of solely within the intended modal.

### What changes do you think we should make to solve the problem?
We should prevent users from focusing outside the modal. One option to do this is to use the `focus-trap-react` library.

We could add the `FocusTrap` to the `ScreenWrapper` with i.g. the `shouldEnableFocusTrap` option. We should probably also specify `fallbackFocus` when there are no focusable items in the modal.

```javascript
<FocusTrap
    active={this.props.shouldEnableFocusTrap}
    focusTrapOptions={{
        fallbackFocus: () => this.fallbackRef.current,
    }}
>
```

We could also set `shouldEnableFocusTrap` to `true` as default, and disable the focus trap when needed (i.g. in the `BaseSidebarScreen`).

This would prevent any undesired interaction with the background and would ensure that all actions remain concentrated within the intended area. Moreover, implementing this solution would **improve the accessibility** of our application.


<details>
<summary>A quick PoC video</summary>

https://github.com/Expensify/App/assets/104823336/bda22588-467c-468f-a44b-99778d8f5e61

</details>

### What alternative solutions did you explore? (Optional)
> But the question still remains, can we implement focus trapping in react navigation? Or it's a flaw in our code?

I believe that doing that in our codebase might be a better solution. The issue here is that we have an RHP displayed as a modal. If the `StackView` would be full-screen, the issue probably wouldn't occur. In the `react-navigation` in the `StackView` we can't focus on the background views, the browser doesn't allow us to do that.

**Comment by kevinksullivan** [comment]
Bump @rushatgabhane

**Comment by DevWizard0000** [comment]
### Proposal
*Please re-state the problem that we are trying to solve in this issue.
Scroll-up when pressing Space bar.

*What is the root cause of that problem?
If the Scroll view get the event of space bar triggered, the scroll view scrolls up the view.

*What changes do you think we should make in order to solve the problem?
We should remove the focus from the scroll view when you are pressing space bar.

*What alternative solutions did you explore? (Optional)
We can focus another component when pressing the space bar or set options to the scroll view.

**Comment by kevinksullivan** [comment]
Reassigning C+

**Comment by kevinksullivan** [comment]
Reassigning

**Comment by kevinksullivan** [comment]
@fedirjh will you be able to take this on?

**Comment by situchan** [comment]
Heads-up: So far no one identified the correct root cause and proposed right solution. This is neither related to navigation nor RHP modals.
This is easily reproducible on Report screen itself:


https://github.com/Expensify/App/assets/108292595/4ca204ce-c1e6-45c7-a9b4-53ae188f7854

**Comment by kosmydel** [comment]
> I don't believe we should apply the focus trap to all modals.

You are right. We need to use it only in the RHP because it doesn't use the `BaseModal` component. 

All other modals which use `BaseModal` already use `ReactNativeModal` component which uses the [`ModalFocusTrap`](https://github.com/necolas/react-native-web/blob/master/packages/react-native-web/src/exports/Modal/ModalFocusTrap.js) from the `react-native-web` library.

We could try to reuse this code instead of adding a new library.

---

> cc @tienifr Could you please update the root cause analysis to address this comment [#15631 (comment)](https://github.com/Expensify/App/issues/15631#issuecomment-1656350799) ?

I think that I identified the root cause correctly in [my proposal](https://github.com/Expensify/App/issues/15631#issuecomment-1641623988):
> Users can focus outside the RHP, allowing them to interact with elements in the background instead of solely within the intended modal.

---

I updated [my proposal](https://github.com/Expensify/App/issues/15631#issuecomment-1641623988).

**Comment by kevinksullivan** [comment]
^ Still working through proposals

**Comment by kosmydel** [comment]
>> make RHP use the Modal component (which might be problematic) which takes care of the focus trap.
> 
> Nope, this doesn’t look like a valid solution to me. This will require another Navigation refactor.

This was just an alternative :) I should have marked it more explicitly.

**Comment by kosmydel** [comment]
> > We could reuse this code instead of adding a new library.
> 
> @kosmydel can you please elaborate on this point, how can we reuse the same code ?

I had an idea that it may be possible to use the same `ModalFocusTrap` as `react-native-web` uses. But after having a look, apparently **it is not exported from the module**. So it may be impossible without changes to that library. If it is preferred solution I can investigate it more.

This is how `ModalFocusTrap` is used in `Modal` in this library ([GitHub link](https://github.com/necolas/react-native-web/blob/ef95fc0c2b40cc0d3743df4b56485a16a8442e5f/packages/react-native-web/src/exports/Modal/index.js#L133)):
```javascript
  <ModalFocusTrap active={isActive}>
    <ModalContent {...rest} >
      {children}
    </ModalContent>
  </ModalFocusTrap>
```

The code from the `react-native-web` is still much lighter than the `react-focus-trap`. But it is missing many props/methods which may be useful in the future.

---

To sum up, adding the `focus-trap` might be the less problematic solution.

**Comment by kevinksullivan** [comment]
@stephanieelliott FYI I am going OOO for a week, but you are on BZ so I won't add anyone else to keep this moving. Thanks!

**Comment by stephanieelliott** [comment]
Hey @fedirjh any thoughts on the additional context provided in [this comment](https://github.com/Expensify/App/issues/15631#issuecomment-1662180433)?

**Comment by stephanieelliott** [comment]
Bump on the above @fedirjh

**Comment by kosmydel** [comment]
Hey @fedirjh, I think you forgot to assign me to the issue. I am starting to work on the PR.

**Comment by stephanieelliott** [comment]
PR is up, awaiting C+ review

**Comment by stephanieelliott** [comment]
PR has been updated and reviewed, this one is moving along in the PR!

**Comment by stephanieelliott** [comment]
Hey @roryabraham, this one is waiting on your PR review!

**Comment by roryabraham** [comment]
Circled back to review the PR, but I heard [through the grapevine](https://expensify.slack.com/archives/C07J32337/p1693250562071879?thread_ts=1692799992.392829&cid=C07J32337) that we are doing a feature freeze starting like now-ish to focus on shipping the features we need in time for some upcoming conferences.

I think that this bug and the associated solution of has a high risk-to-value ratio so should be put on HOLD according to the feature freeze (which should be publicly announced very soon).

**Comment by stephanieelliott** [comment]
The feature freeze has[ now been announced](https://expensify.slack.com/archives/C07J32337/p1693347966078669?thread_ts=1692799992.392829&cid=C07J32337). I agree, let's put this on hold. Based on the post, ETA to resume is ~11th September.

**Comment by stephanieelliott** [comment]
The original PR (#24316) was reverted due to a few bugs, the new PR for this issue is https://github.com/Expensify/App/pull/27670

Still on hold — now held on #27938

**Comment by stephanieelliott** [comment]
Taking off hold, the blocking PR was merged! PR for this issue is https://github.com/Expensify/App/pull/27670, currently under review.

**Comment by stephanieelliott** [comment]
Still actively being reviewed

**Comment by stephanieelliott** [comment]
Still actively being reviewed

**Comment by stephanieelliott** [comment]
Some failing tests on the PR, @roryabraham looking for your input on the PR: https://github.com/Expensify/App/pull/27670

**Comment by thienlnam** [comment]
This PR was reverted - not ready for payment yet

**Comment by stephanieelliott** [comment]
Cross-posting update from reverted PR https://github.com/Expensify/App/pull/27670

> we're reverting this PR since it caused the following blockers:
> 
> https://github.com/Expensify/App/issues/31687
> https://github.com/Expensify/App/issues/31725
> https://github.com/Expensify/App/issues/31734
> Could you please work on the PR and a fix for the issues above?

@kosmydel is planning to work on this next week

**Comment by kosmydel** [comment]
Hey, 
I've had a look at this issue. I've fixed two of three regressions (here is a draft of the new PR). The third one is reproducible on main, and [being worked on](https://github.com/Expensify/App/issues/31734#issuecomment-1849833018) by @kowczarz.

---

However, I've found out that [we don't care about the tab functionality at all](https://expensify.slack.com/archives/C049HHMV9SM/p1700234377291559?thread_ts=1700217396.857389&cid=C049HHMV9SM). We can try to go with the easier path, that uses focus trap **only** on the problematic screens from the original issue (that is e.g. `ProfilePage`). In that way, we would affect a few screens, instead of the whole app.

What do you think about this path? This would save us from another possible regression, as adding a focus trap is risky, especially since it affects almost every screen in the App (and it is almost impossible for us to test every screen). The downside is, that the focus trap would not be working in any other place.

---

So we can either:
1. Add a focus trap to every screen, risking another regression.
2. Safer option: add a trap to selected screens, but the tab navigation will still be broken in most of the screens.

cc @fedirjh, @luacmartins, @roryabraham

**Comment by luacmartins** [comment]
I think we should go with 1 and handle focus trap across the whole app, we already ran a regression test and found a few so I think as long as we fix those we should be ok.

**Comment by kosmydel** [comment]
> I think we should go with 1 and handle focus trap across the whole app, we already ran a regression test and found a few so I think as long as we fix those we should be ok.

Sure! I've prepared [a PR](https://github.com/Expensify/App/pull/32800) with fixes. One _regression_ wasn't from my PR and will be fixed in [this PR](https://github.com/Expensify/App/pull/32815).

I will test it and make it ready for a review tomorrow.

**Comment by roryabraham** [comment]
Looks like we're currently HOLDing for https://github.com/Expensify/App/pull/32815

**Comment by roryabraham** [comment]
Also, I'm going to go ahead and make this lower-priority issue into a weekly

**Comment by kosmydel** [comment]
> Looks like we're currently HOLDing for #32815
@roryabraham linked PR fixes regression which wasn't coming from my changes (but it was linked to my PR anyway), so I think we don't need to hold.

**Comment by stephanieelliott** [comment]
https://github.com/Expensify/App/pull/32815 was merged, so I think we can proceed here either way. Conflicts have been resolved in https://github.com/Expensify/App/pull/32800, currently waiting on @fedirjh's review!

**Comment by stephanieelliott** [comment]
PR is being actively reviewed!

**Comment by stephanieelliott** [comment]
PR is reviewed, but we're going to do the QA before merging.

**Comment by stephanieelliott** [comment]
Kicked off QA [here](https://expensify.slack.com/archives/C9YU7BX5M/p1705209423974619)

**Comment by stephanieelliott** [comment]
QA testing was completed ([notes](https://github.com/Expensify/App/pull/32800#issuecomment-1908867344)), no issues 🎉

**Comment by stephanieelliott** [comment]
PR is under review

**Comment by roryabraham** [comment]
@kosmydel reports this was unknowingly fixed by #ideal-nav. I wasn't able to reproduce this anymore. I think we should close it out?

**Comment by JmillsExpensify** [comment]
Ha, well would you look at that!

**Comment by stephanieelliott** [comment]
Love it when that happens! Ok, given that we had 3 PRs reviewed (2 of which were merged),  I think payment is due here however the regression penalty would apply. Some quick math: 

Original job price $1000 
- 50% regression penalty PR 1 (#24316) = $500
- 50% regression penalty PR 2 (#27670) = $250

---

That would make the payout summary: 

 - Contributor: @kosmydel N/A, expert contributor 
 - Contributor+: @fedirjh $250 via Upwork (offer extended, please accept)
 
 Upwork job is here: https://www.upwork.com/jobs/~0112187286e2fd23a2

**Comment by stephanieelliott** [comment]
All paid!

**Comment by kosmydel** [comment]
> @kosmydel reports this was unknowingly fixed by #ideal-nav. I wasn't able to reproduce this anymore. I think we should close it out?

Hey, good to know that the issue doesn't occur anymore.

However, I want to confirm, if we don't want to introduce the FocusTrap anyway. It did not only meant to solve this issue, but also improved the accessibility and tab navigation on the web/desktop.

cc @luacmartins, @roryabraham

**Comment by roryabraham** [comment]
> It did not only meant to solve this issue, but also improved the accessibility and tab navigation on the web/desktop.

@kosmydel can you be more specific about what accessibility problem it solved?

**Comment by kosmydel** [comment]
> @kosmydel can you be more specific about what accessibility problem it solves?

Tab navigation is used together with e.g. VoiceOver on macOS to help people with visual disabilities. 

Currently, if the e.g. RHP modals don't support focus trap, tab navigation is almost impossible (or at least cumbersome). We are "tabbing" outside the modal, which is probably not what a user expects after pressing the button that opens it. The navigation order is not correct.

[Here is more info about navigation order.](https://webaim.org/techniques/keyboard/)
